### DISCUSSION TITLE
Isso client is stripping newlines

### DISCUSSION THREAD

**Discussion by monodot** [posativ/isso#856] [General]
Hello isso'ers!

I noticed that the client-side JS strips newlines from the output in `isso/js/app/template.js`.

https://github.com/posativ/isso/blob/178912d4426155201a67e0a95cff0ac201494d28/isso/js/app/template.js#L67-L68

This means that comments containing fenced code blocks -- which are rendered server-side as `<pre><code>some code here\nanother new line</code></pre>` -- have their `\n` newline characters ignored when the client-side JS renders, and so the whole output gets rendered on one line. This makes code blocks hard to read.

I'd quite like to keep these newline characters when the comment is rendered. I tried just commenting that newline-stripping line of code, and everything seems to work. Is there a reason for this line being there?

Thanks!

**Comment by ix5** [comment]
See https://github.com/posativ/isso/pull/857

**Reply by ix5** [reply]
@monodot I hope https://github.com/posativ/isso/pull/857 fixes things to your satisfaction.

Would be great if you could contribute more test cases to the tests - simply copy-paste the block containing the `text` dict item from https://github.com/posativ/isso/blob/6f429851bf1af2d8f31abac0b4bd843c51e76545/isso/js/tests/unit/template-comment-newlines.test.js#L59
and fill in your own input - then check that rendered output in https://github.com/posativ/isso/blob/6f429851bf1af2d8f31abac0b4bd843c51e76545/isso/js/tests/unit/__snapshots__/template-comment-newlines.test.js.snap#L3-L10
matches what you expect.
